tonic_validate/metrics/binary_metric.py

```diff
@@ -0,0 +1,23 @@
+import logging
+from typing import Callable
+
+from tonic_validate.classes.llm_response import LLMResponse
+from tonic_validate.metrics.metric import Metric
+from tonic_validate.services.openai_service import OpenAIService
+
+logger = logging.getLogger()
+
+
+class BinaryMetric(Metric):
+    @property
+    def name(self) -> str:
+        return self._name
+
+    def __init__(
+        self, name: str, callback: Callable[[LLMResponse, OpenAIService], bool]
+    ):
+        self._name = name
+        self.callback = callback
+
+    def score(self, llm_response: LLMResponse, openai_service: OpenAIService) -> float:
```

Comment:
We need some way to be able to use these metrics without passing an `OpenAIService` for the metrics that do not require an `OpenAIService`. My suggestion would be to make this parameter optional and default to None.

Reply:
All the metrics are called by our `ValidateScorer`. If we allow for it to be None, then we'd need to change the logic in the scorer to not pass it for metrics that don't need it. That becomes harder to do when people are writing their own metrics (e.g. a custom binary metric) as we don't know if they're using OpenAI or not. Sure we could have them set something like `self.uses_openai = True`, but that's a bad pattern since it relies on the user remembering to change it. It's easier just to pass it in regardless in which case `OpenAIService` will never be `None` anyhow. Plus, if we set it to be optional, then we will have to rewrite all of our existing code to check that it's not None in order to satisfy typechecking.

Reply:
The case I'm thinking of is when a user is using one of the metrics outside of `ValidateScorer`. In that case, they should not need to instantiate and pass an `OpenAIService` when it's not needed.

If it's optional then `ValidateScorer` can pass all the metrics its `OpenAIService` without any issues.

Reply:
That's not worth it I think because that's a really niche use case and it would lead to us having to add checks everywhere in the code for it (as well as checks when the person writes their own metrics)

Reply:
I think whether a metric actually needs an LLM evaluator is an important distinction to make and should be reflected in the metric. I can imagine users thinking, I want to create a bunch of tests that use metrics that do not use an LLM evaluator (because the tests are way cheaper without an LLM evaluator). Then they don't need an `OpenAIService` and passing a model evaluator to `ValidateScorer` is unnecessary.

What do you think about putting the logic into the classes, so we would have a metric base class that uses LLM assisted evaluation and a metric base class that does not use LLM assisted evaluation? In the metrics that do not use LLM assisted evaluation, they would not need an `OpenAIService`.

Reply:
Yeah, I can add a subclass in for this. That would make sense, but I'll probably do it in a separate PR since it's a bit out of scope